src/configuration/rulesForFile.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as minimatch from "minimatch";
+import * as path from "path";
+import { IOptions, RuleSeverity } from "../index";
+import { IConfigurationFile, RuleMap } from "./configurationFile";
+
+const IS_JS = /\.jsx?$/i;
+
+/**
+ * A file to lint. Stores the isJS property to avoid extra regex tests.
+ */
+interface SourceFile {
+    path: string;
+    isJS: boolean;
+}
+
+export function rulesForFile(config: IConfigurationFile, filePath: string): IOptions[] {
```

Comment:
This is where the majority of the new functionality is:
  * Getting `rules` of `jsRules` based on the file name
  * Using `extends` to add additional rules
  * Looking at `overrides` for override rules